src/main/java/org/jenkinsci/plugins/workflow/steps/MatrixStep.java

```diff
@@ -0,0 +1,84 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import com.google.common.collect.Lists;
+import hudson.Extension;
+import org.jenkinsci.plugins.workflow.cps.steps.ingroovy.GroovyStep;
+import org.jenkinsci.plugins.workflow.cps.steps.ingroovy.GroovyStepDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MatrixStep extends GroovyStep {
+    private Map<String,List<Object>> axes;
+
+    @DataBoundConstructor
+    public MatrixStep(@Nonnull Map<String,List<Object>> axes) {
+        this.axes = axes;
+    }
+
+    public Map<String,List<Object>> getAxes() {
```

Comment:
But these _cannot_ all work in an actual `Step`. You can only use `String`s, since that is all databinding offers.